### DISCUSSION TITLE
注入更新问题

### DISCUSSION THREAD

**Discussion by timBen9527** [midwayjs/midway#3362] [Q&A] [answered]
### 🐛 Describe the bug（描述 bug）

我在项目启动时获取了地区的mmdb，但存在一个问题，每周会更新mmdb文件，更新后我该如何更新MMDBReader对象

```
async onReady(container: IMidwayContainer) {
    this.dbBuffer = fs.readFileSync(path.join(`${this.appDir}/Country.mmdb`));
    this.reader = Reader.openBuffer(this.dbBuffer);
    container.registerObject('MMDBUtils', {
      updateMMDB: this.updateMMDB,
      MMDBReader: this.reader
    });
  }
```

### Midway Versions（Midway 版本）

✓ @midwayjs/faas-typings(not installed)
✓ @midwayjs/fc-starter(not installed)
✓ @midwayjs/serverless-http-parser(not installed)
✓ @midwayjs/async-hooks-context-manager(3.12.3)
✓ @midwayjs/axios(not installed)
✓ @midwayjs/bootstrap(3.12.5)
✓ @midwayjs/bull(not installed)
✓ @midwayjs/bull-board(not installed)
✓ @midwayjs/cache(3.12.3)
✓ @midwayjs/captcha(not installed)
✓ @midwayjs/casbin(not installed)
✓ @midwayjs/casbin-redis-adapter(not installed)
✓ @midwayjs/casbin-typeorm-adapter(not installed)
✓ @midwayjs/code-dye(not installed)
✓ @midwayjs/consul(3.12.7)
✓ @midwayjs/core(3.12.3)
✓ @midwayjs/cos(not installed)
✓ @midwayjs/cron(not installed)
✓ @midwayjs/cross-domain(3.12.7)
✓ @midwayjs/decorator(3.12.3)
✓ @midwayjs/etcd(not installed)
✓ @midwayjs/express-session(not installed)
✓ @midwayjs/faas(not installed)
✓ @midwayjs/grpc(not installed)
✓ @midwayjs/http-proxy(not installed)
✓ @midwayjs/i18n(3.12.7)
✓ @midwayjs/info(3.12.7)
✓ @midwayjs/jwt(not installed)
✓ @midwayjs/kafka(not installed)
✓ @midwayjs/leoric(not installed)
✓ @midwayjs/mikro(not installed)
✓ @midwayjs/mock(3.12.3)
✓ @midwayjs/mongoose(not installed)
✓ @midwayjs/oss(not installed)
✓ @midwayjs/otel(not installed)
✓ @midwayjs/passport(not installed)
✓ @midwayjs/process-agent(not installed)
✓ @midwayjs/prometheus(not installed)
✓ @midwayjs/prometheus-socket-io(not installed)
✓ @midwayjs/rabbitmq(not installed)
✓ @midwayjs/redis(not installed)
✓ @midwayjs/security(not installed)
✓ @midwayjs/sequelize(not installed)
✓ @midwayjs/session(3.12.3)
✓ @midwayjs/socketio(not installed)
✓ @midwayjs/static-file(3.12.5)
✓ @midwayjs/swagger(3.12.7)
✓ @midwayjs/tablestore(not installed)
✓ @midwayjs/tags(not installed)
✓ @midwayjs/typegoose(not installed)
✓ @midwayjs/typeorm(3.12.3)
✓ @midwayjs/upload(3.12.7)
✓ @midwayjs/validate(3.12.7)
✓ @midwayjs/version(3.12.7)
✓ @midwayjs/view(3.12.3)
✓ @midwayjs/view-ejs(3.12.3)
✓ @midwayjs/view-nunjucks(not installed)
✓ @midwayjs/web(not installed)
✓ @midwayjs/express(not installed)
✓ @midwayjs/koa(3.12.3)
✓ @midwayjs/ws(not installed)
**********************************************************************
>> Check complete, all versions are healthy.
**********************************************************************

**Comment by czy88840616** [comment, marked as answer]
使用订阅模型。https://midwayjs.org/docs/data_listener
